cognite/client/_api/data_modeling/instances.py

```diff
@@ -738,31 +740,73 @@ def search(
         res = self._post(url_path=self._RESOURCE_PATH + "/search", json=body)
         return list_cls.load(res.json()["items"], cognite_client=None)
 
+    @overload
+    def aggregate(
+        self,
+        view: ViewId,
+        aggregates: MetricAggregation | dict,
+        group_by: None = None,
+        instance_type: Literal["node", "edge"] = "node",
+        query: str | None = None,
+        properties: str | MutableSequence[str] | None = None,
+        filter: Filter | None = None,
+        limit: int = DEFAULT_LIMIT_READ,
+    ) -> AggregatedNumberedValue:
+        ...
+
+    @overload
+    def aggregate(
+        self,
+        view: ViewId,
+        aggregates: Sequence[MetricAggregation | dict],
+        group_by: None = None,
+        instance_type: Literal["node", "edge"] = "node",
+        query: str | None = None,
+        properties: str | MutableSequence[str] | None = None,
+        filter: Filter | None = None,
+        limit: int = DEFAULT_LIMIT_READ,
+    ) -> list[AggregatedNumberedValue]:
+        ...
+
+    @overload
     def aggregate(
         self,
         view: ViewId,
         aggregates: MetricAggregation | dict | Sequence[MetricAggregation | dict],
+        group_by: str | MutableSequence[str],
```

Comment:
Same note on MutableSequence as in the other PR. We'll need to introduce smth like `SequenceNotStr` to property implement these overloads.